src/main/java/net/pms/device/DeviceProtocolInfo.java

```diff
@@ -0,0 +1,430 @@
+	 */
+	public boolean contains(DLNAProtocolInfo protocolInfo) {
+		setsLock.readLock().lock();
+		try {
+			return protocolInfoSet.contains(protocolInfo);
+		} finally {
+			setsLock.readLock().unlock();
+		}
+	}
+
+	/**
+	 * Returns an array containing all of the elements in an unspecified order.
+	 * <p>
+	 * The returned array will be "safe" in that no references to it are
+	 * maintained. (In other words, this method must allocate a new array). The
+	 * caller is thus free to modify the returned array.
+	 *
+	 * @return An array containing all of the elements.
+	 */
+	public DLNAProtocolInfo[] toArray() {
+		setsLock.readLock().lock();
+		try {
+			return protocolInfoSet.toArray(new DLNAProtocolInfo[protocolInfoSet.size()]);
+		} finally {
+			setsLock.readLock().unlock();
+		}
+	}
+
+
+	/**
+	 * Returns {@code true} if this contains all of the elements in the
+	 * specified collection.
+	 *
+	 * @param collection a {@link Collection} to be checked for containment.
+	 * @return {@code true} if this collection contains all of the elements in
+	 *         {@code collection}.
+	 *
+	 * @see #contains(DLNAProtocolInfo))
+	 */
+	public boolean containsAll(Collection<DLNAProtocolInfo> collection) {
+		setsLock.readLock().lock();
+		try {
+			return protocolInfoSet.containsAll(collection);
+		} finally {
+			setsLock.readLock().unlock();
+		}
+	}
+
+
+	/**
+	 * Removes all elements.
+	 */
+	public void clear() {
+		setsLock.writeLock().lock();
+		try {
+			protocolInfoSet.clear();
+			imageProfileSet.clear();
+		} finally {
+			setsLock.writeLock().unlock();
+		}
+	}
+
+	/**
+	 * Ensures that this contains the specified element. Returns {@code true} if
+	 * this changed as a result of the call. Returns {@code false} this already
+	 * contains the specified element.
+	 *
+	 * @param protocolInfo element whose presence is to be ensured.
+	 * @return {@code true} if this changed as a result of the call.
+	 */
+	public boolean add(DLNAProtocolInfo protocolInfo) {
+		setsLock.writeLock().lock();
+		try {
+			if (protocolInfoSet.add(protocolInfo)) {
+				updateImageProfiles();
+				return true;
+			}
+			return false;
+		} finally {
+			setsLock.writeLock().unlock();
+		}
+	}
+
+	/**
+	 * Adds all of the elements in the specified collection.
+	 *
+	 * @param collection a {@link Collection} containing elements to be added.
+	 * @return {@code true} if this changed as a result of the call.
+	 *
+	 * @see #add(DLNAProtocolInfo)
+	 */
+	public boolean addAll(Collection<? extends DLNAProtocolInfo> collection) {
+		setsLock.writeLock().lock();
+		try {
+			if (protocolInfoSet.addAll(collection)) {
+				updateImageProfiles();
+				return true;
+			}
+			return false;
+		} finally {
+			setsLock.writeLock().unlock();
+		}
+	}
+
+	/**
+	 * Removes a single instance of the {@link DLNAProtocolInfo}, if it is
+	 * present. Returns {@code true} if this contained the specified element (or
+	 * equivalently, if this changed as a result of the call).
```

Comment:
I was referring to the conjugaison time used in that sentence.
"has changed" or "has been changed" ? instead of "changed".
That's said, as you know my weak english level that could be only a bad understanding from my side.